state/remoteapplication.go

```diff
@@ -29,12 +29,13 @@ type RemoteApplication struct {
 
 // remoteApplicationDoc represents the internal state of a remote application in MongoDB.
 type remoteApplicationDoc struct {
-	DocID         string              `bson:"_id"`
-	Name          string              `bson:"name"`
-	URL           string              `bson:"url"`
-	Endpoints     []remoteEndpointDoc `bson:"endpoints"`
-	Life          Life                `bson:"life"`
-	RelationCount int                 `bson:"relationcount"`
+	DocID           string              `bson:"_id"`
```

Comment:
Oh, and the only reason the fields need to be exported is so that the bson serialisation can work. The struct itself is not exported so it really is an internal thing.